plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaMetadata.java

```diff
@@ -294,10 +300,20 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
                 table.getMessageDataSchemaLocation(),
                 table.getKeySubject(),
                 table.getMessageSubject(),
-                actualColumns,
+                columns.stream()
+                        .map(KafkaColumnHandle.class::cast)
+                        .collect(toImmutableList()),
                 TupleDomain.none());
     }
 
+    @Override
+    public boolean supportsMissingColumnsOnInsert(ConnectorTableHandle tableHandle)
+    {
+        KafkaTableHandle table = (KafkaTableHandle) tableHandle;
+        return dispatchingRowEncoderFactory.supportsMissingColumns(table.getKeyDataFormat()) &&
```

Comment:
This means that defaults won't work for records with different formats of keys and values. ie. AVRO value + RAW key. A test covering such cases would be useful.

Reply:
Good catch. This is gonna result in inconsistent behavior. How do we address this? 
cc @kokosing

Reply:
Exactly, we currently don't have a way to specify which column to add null or which one to skip. It works for only if encoder of key and message both supports missing column.

Reply:
The way out of this would be to support missing columns for all formats, ie move the responsibility of NULL-ifying missing columns to the row encoder. That would make raw and csv encoder more complicated, but would save the feature consistency of avro/avro and raw/avro cases. (Actually, do Kafka users use Avro as the key encoding? or only for the data part?)

Reply:
Can you please add a tests for raw/avro and avro/row?